# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/youssefayman8070-ux/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I selected Lane 2: Refresh / Content Opportunity Scoring. I want to explore how safe content and performance signals can help identify which pages should be reviewed first for refreshing, improving, protecting, or monitoring. I chose this lane because it connects the available data to a clear operational decision and can produce a transparent ranked queue with understandable reason codes. The starter notebooks also showed that page trends cannot be explained by one simple rule, so a careful data-driven approach may provide better prioritization.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision:
Decide which pages should be reviewed first for content refresh.

Action:
The SEO or content team reviews the highest-priority pages and updates or improves them before spending time on lower-priority pages.

Cost of a wrong recommendation:
If the recommendation is wrong, the team may waste time updating pages that do not need changes or ignore pages that actually need attention, reducing the impact of SEO efforts.

Why ML helps:
Many signals affect page performance at the same time. Machine learning can combine these signals and help prioritize pages more consistently than simple manual rules.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import os
import subprocess
import pandas as pd

REPO_URL = "https://github.com/youssefayman8070-ux/flyrank-ml-internship.git"
REPO_DIR = "flyrank-ml-internship"

if not os.path.exists(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL], check=True)

csv_path = f"{REPO_DIR}/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(csv_path)

print("Number of pages:", len(df))
print("Average CTR:", round(df["ctr"].mean(), 4))
print("Average impressions:", round(df["impressions_90d"].mean(), 2))

Number of pages: 30000
Average CTR: 0.5107
Average impressions: 5200.37


The starter dataset contains 30,000 pages.

The average CTR is 0.5107.

The average number of impressions is 5,200.37.

These numbers show that the dataset is large enough to study page performance and build a practical page-prioritization approach for content refresh decisions.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will describe observed relationships in the available data and support SEO decisions.

It will not prove that any signal causes better Google rankings or predict Google's algorithm.

Any recommendations should be treated as decision support based on observed patterns, not as causal evidence.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.